kubera/dmaas/credentials/add_credential_test.go

```diff
@@ -0,0 +1,116 @@
+package credentials
+
+import (
+	"os"
+	"testing"
+)
+
+func TestCredentialAddition(t *testing.T) {
+
+	const (
+		awsAccessID       = "AWS_ACCESS_ID"
+		awsSecretKey      = "AWS_SECRET_KEY"
+		cloudianAccessID  = "CLOUDIAN_ACCESS_ID"
+		cloudianSecretKey = "CLOUDIAN_SECRET_KEY"
+		minioAccessID     = "MINIO_ACCESS_ID"
+		minioSecretKey    = "MINIO_SECRET_KEY"
+	)
+	var tests = map[string]struct {
+		cloud     string
+		name      string
+		accessID  string
+		secretKey string
+		IsError   bool
+	}{
+		"aws bucket with all details": {
+			cloud:     "aws",
+			name:      "aws-dmaas-test",
+			accessID:  os.Getenv(awsAccessID),
+			secretKey: os.Getenv(awsSecretKey),
+		},
+		"aws bucket without name": {
+			cloud:     "aws",
+			accessID:  os.Getenv(awsAccessID),
+			secretKey: os.Getenv(awsSecretKey),
+			IsError:   true,
+		},
+		"aws bucket withot username ": {
+			cloud:     "aws",
+			name:      "aws-dmaas-test",
+			secretKey: os.Getenv(awsSecretKey),
+			IsError:   true,
+		},
+		"aws bucket without password": {
+			cloud:    "aws",
+			name:     "aws-dmaas-test",
+			accessID: os.Getenv(awsAccessID),
+			IsError:  true,
+		},
+		"cloudian bucket with all details": {
+			cloud:     "cloudian",
+			name:      "cloudian-dmaas-test",
+			accessID:  os.Getenv(cloudianAccessID),
+			secretKey: os.Getenv(cloudianSecretKey),
+		},
+		"cloudian bucket withot name": {
+			cloud:     "cloudian",
+			accessID:  os.Getenv(cloudianAccessID),
+			secretKey: os.Getenv(cloudianSecretKey),
+			IsError:   true,
+		},
+		"cloudian bucket without username": {
```

Comment:
keep either username or secretkey. Whichever is more readable & understood by the teams.